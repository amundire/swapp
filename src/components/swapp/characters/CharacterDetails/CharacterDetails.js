import React from 'react';
import Starship from '../../starships/Starship/Starship';

const CharacterDetails = ({data}) => {
    return (
        <section id="characterDetails">
            <div className="characterDetailsLeft">
                <div className="characterDetailsName">{data.person.Name}</div>
                <img src={data.person.image}></img>
                <div className="characterHeight">{data.person.height}</div>
                <div className="characterMass">{data.person.mass}</div>
            </div>
            <div className="characterDetailsRight">
                <p className="starshipsTitle">Piloted Starships</p>
                <ul>
                    {data.person.starships.edges.map(({ node }) =>
                        <Starship id={node.id} image={node.image} name={node.name} key={node.id} />
                        )}
                </ul>
            </div>
        </section>
    )
}

export default CharacterDetails;