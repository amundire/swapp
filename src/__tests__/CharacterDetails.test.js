import React from 'react';
import { shallow } from 'enzyme';
import CharacterDetails from '../components/swapp/characters/CharacterDetails/CharacterDetails.js';

describe('<CharacterDetails />', () => {
    it('renders without crashing', () => {
        const data = {
            person: {
                id: 'people.1',
                name: 'Luke Skywalker',
                height: 172,
                mass: 77,
                image: 'https://links.gunaxin.com/content/images/post_images/Luke_Skywalker_039_s_Original_039_Star_Wars_039_Lightsaber_Is_Going_Up_For_Auction_1544116268_4528.jpg',
                homeworld: {
                    name: 'Tatooine',
                },
                species: {
                    name: 'Human',
                },
                starships: {
                    edges: [],
                },
            },
        },
            wrapper = shallow(<CharacterDetails data={data} />);
        expect(wrapper).toMatchSnapshot();
    });
});
