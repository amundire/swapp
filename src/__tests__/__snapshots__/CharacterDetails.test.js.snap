// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CharacterDetails /> renders without crashing 1`] = `
<section
  id="characterDetails"
>
  <p
    className="characterDetailsName"
  >
    Luke Skywalker
  </p>
  <div
    className="characterDetailsDivisor"
  >
    <div
      className="characterDetailsLeft"
    >
      <p
        className="characterDetailsName"
      >
        Luke Skywalker
      </p>
      <img
        src="https://links.gunaxin.com/content/images/post_images/Luke_Skywalker_039_s_Original_039_Star_Wars_039_Lightsaber_Is_Going_Up_For_Auction_1544116268_4528.jpg"
      />
      <div
        className="characterHeight"
      >
        Height: 
        172
      </div>
      <div
        className="characterMass"
      >
        Weight: 
        77
      </div>
      <div
        className="characterMass"
      >
        Species: 
        Human
      </div>
      <div
        className="characterMass"
      >
         
        Tatooine
      </div>
    </div>
    <div
      className="characterDetailsRight"
    >
      <p
        className="starshipsTitle"
      >
        Piloted Starships
      </p>
      <ul />
    </div>
  </div>
</section>
`;
